test/test_net_util.cc

```diff
@@ -1,60 +1,59 @@
-#include <stdint.h>
-#include <stdio.h>
-#include <string.h>
-#include <stdlib.h>
-
-#include <gtest/gtest.h>
-
-#include "net_util.h"
-
-TEST(dns, resolve_local) {
-    int32_t error = 0;
-    char *ip = dns_resolve(NULL, &error);
-    EXPECT_EQ(error, 0);
-    EXPECT_NE(ip, nullptr);
-    if (ip) {
-        EXPECT_TRUE(strstr(ip, ".") != NULL);
-        free(ip);
-    }
-}
-
-TEST(dns, resolve_google) {
-    int32_t error = 0;
-    char *ip = dns_resolve("google.com", &error);
-    EXPECT_EQ(error, 0);
-    EXPECT_TRUE(ip != NULL);
-    if (ip) {
-        EXPECT_TRUE(strstr(ip, ".") != NULL);
-        EXPECT_TRUE(strstr(ip, ";") == NULL);
-        free(ip);
-    }
-}
-
-TEST(dns, resolve_bad) {
-    int32_t error = 0;
-    char *ip = dns_resolve("hopefully-doesnt-exist.com", &error);
-    EXPECT_EQ(ip, nullptr);
-    EXPECT_NE(error, 0);
-}
-
-TEST(dns_ex, resolve_local) {
-    int32_t error = 0;
-    char *ips = dns_resolve_ex(NULL, &error);
-    EXPECT_EQ(error, 0);
-    EXPECT_TRUE(ips != NULL);
-    if (ips) {
-        EXPECT_TRUE(strstr(ips, ".") != NULL || strstr(ips, ":") != NULL);
-        free(ips);
-    }
-}
-
-TEST(dns_ex, resolve_google) {
-    int32_t error = 0;
-    char *ips = dns_resolve_ex("google.com", &error);
-    EXPECT_EQ(error, 0);
-    EXPECT_TRUE(ips != NULL);
-    if (ips) {
-        EXPECT_TRUE(strstr(ips, ".") != NULL || strstr(ips, ":") != NULL);
-        free(ips);
-    }
-}
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include <gtest/gtest.h>
+
+#include "net_util.h"
+
+TEST(net_util, my_ip_address) {
+    char *address = my_ip_address();
+    EXPECT_TRUE(address != NULL);
+    if (address) {
+        EXPECT_TRUE(strstr(address, ".") != NULL);

```

Comment:
`nullptr` in C++.
Why not
```suggestion
        EXPECT_NE(strstr(address, "."), nullptr);
```
IMHO its output is more verbose.

I would use `strchr()` or `::testing::Contains()` there.